dex/testing/doge/harness.sh

```diff
@@ -0,0 +1,210 @@
+#!/usr/bin/env bash
+SYMBOL="doge"
+DAEMON="dogecoind"
+CLI="dogecoin-cli"
+RPC_USER="user"
+RPC_PASS="pass"
+ALPHA_LISTEN_PORT="23764"
+BETA_LISTEN_PORT="23765"
+ALPHA_RPC_PORT="23766"
+BETA_RPC_PORT="23767"
+ALPHA_MINING_ADDR="mjzB71SzfAi8BwNvobPZ983d8vCdxuQD2i"
+BETA_MINING_ADDR="mwL7ypWCMEhBhGcsqWNwkvAasuXP3duXQk"
+
+set -ex
+NODES_ROOT=~/dextest/${SYMBOL}
+rm -rf "${NODES_ROOT}"
+SOURCE_DIR=$(pwd)
+
+ALPHA_DIR="${NODES_ROOT}/alpha"
+BETA_DIR="${NODES_ROOT}/beta"
+HARNESS_DIR="${NODES_ROOT}/harness-ctl"
+
+echo "Writing node config files"
+mkdir -p "${HARNESS_DIR}"
+
+WALLET_PASSWORD="abc"
+
+ALPHA_CLI_CFG="-rpcport=${ALPHA_RPC_PORT} -regtest=1 -rpcuser=user -rpcpassword=pass"
+
+BETA_CLI_CFG="-rpcport=${BETA_RPC_PORT} -regtest=1 -rpcuser=user -rpcpassword=pass"
+
+# DONE can be used in a send-keys call along with a `wait-for btc` command to
+# wait for process termination.
+DONE="; tmux wait-for -S ${SYMBOL}"
+WAIT="wait-for ${SYMBOL}"
+
+SESSION="${SYMBOL}-harness"
+
+SHELL=$(which bash)
+
+################################################################################
+# Load prepared wallet if the files exist.
+################################################################################
+
+mkdir -p ${ALPHA_DIR}/regtest
+cp ${SOURCE_DIR}/alpha_wallet.dat ${ALPHA_DIR}/regtest/wallet.dat
+mkdir -p ${BETA_DIR}/regtest
+cp ${SOURCE_DIR}/beta_wallet.dat ${BETA_DIR}/regtest/wallet.dat
+
+cd ${NODES_ROOT} && tmux new-session -d -s $SESSION $SHELL
+
+################################################################################
+# Write config files.
+################################################################################
+
+# These config files aren't actually used here, but can be used by other
+# programs. I would use them here, but bitcoind seems to have some issues
+# reading from the file when using regtest.
+
+cat > "${ALPHA_DIR}/alpha.conf" <<EOF
+rpcuser=user
+rpcpassword=pass
+datadir=${ALPHA_DIR}
+txindex=1
+port=${ALPHA_LISTEN_PORT}
+regtest=1
+rpcport=${ALPHA_RPC_PORT}
+EOF
+
+cat > "${BETA_DIR}/beta.conf" <<EOF
+rpcuser=user
+rpcpassword=pass
+datadir=${BETA_DIR}
+txindex=1
+regtest=1
+rpcport=${BETA_RPC_PORT}
+EOF
+
+################################################################################
+# Start the alpha node.
+################################################################################
+
+tmux rename-window -t $SESSION:0 'alpha'
+tmux send-keys -t $SESSION:0 "set +o history" C-m
+tmux send-keys -t $SESSION:0 "cd ${ALPHA_DIR}" C-m
+echo "Starting simnet alpha node"
+tmux send-keys -t $SESSION:0 "${DAEMON} -rpcuser=user -rpcpassword=pass \
+  -rpcport=${ALPHA_RPC_PORT} -datadir=${ALPHA_DIR} \
+  -debug=rpc -debug=net -debug=mempool -debug=walletdb -debug=addrman -debug=mempoolrej \
+  -whitelist=127.0.0.0/8 -whitelist=::1 \
+  -txindex=1 -regtest=1 -port=${ALPHA_LISTEN_PORT} -fallbackfee=0.00001 \
+  -printtoconsole; tmux wait-for -S alpha${SYMBOL}" C-m
```

Comment:
Madness that dogecoind requires `printtoconsole`.  I killed it several times in the last week or so thinking it was hung

Reply:
Zcash too, but for different reasons.